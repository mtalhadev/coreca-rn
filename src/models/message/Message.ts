import { ReactionType } from '../reaction/Reaction';
import { ReactionListCLType, ReactionListType, toReactionListCLType } from '../reaction/ReactionListType';
import { ReadType } from '../read/Read';
import { ReadListCLType, ReadListType, toReadListCLType } from '../read/ReadListType';
import { RoomCLType, RoomType, toRoomCLType } from '../room/Room';
import { ThreadHeadCLType, ThreadHeadType, toThreadHeadCLType } from '../threadHead/ThreadHead';
import { toWorkerCLType, WorkerCLType, WorkerType } from '../worker/Worker';
import { CommonCLType, CommonModel, Create, ReplaceAnd, toCommonCLType, Update } from '../_others/Common';
import { ID } from '../_others/ID';
import { GetOptionObjectType, GetOptionParam } from '../_others/Option';


export type MessageEnumType = 'text' | 'picture' | 'movie' | 'note'

export type MessageModel = Partial<{
    messageId: ID;
    roomId: ID;
    threadId: ID;
    isThreadStart: boolean;
    extraRoomId: ID;
    index: number;
    workerId: ID;
    message: string;
    messageType: MessageEnumType;
    replyId: ID;
    attachmentUrl: string;
    sAttachmentUrl: string;
    xsAttachmentUrl: string;
    readCount: number;
    updateCount: number;
}> &
    CommonModel;

export const initMessage = (message: Create<MessageModel> | Update<MessageModel>): Update<MessageModel> => {
    const newMessage: Update<MessageModel> = {
        messageId: message.messageId,
        roomId: message.roomId,
        threadId: message.threadId,
        isThreadStart: message.isThreadStart,
        extraRoomId: message.extraRoomId,
        index: message.index,
        workerId: message.workerId,
        message: message.message,
        messageType: message.messageType,
        replyId: message.replyId,
        attachmentUrl: message.attachmentUrl,
        sAttachmentUrl: message.sAttachmentUrl,
        xsAttachmentUrl: message.xsAttachmentUrl,
        readCount: message.readCount,
        updateCount: message.updateCount,
    };
    return newMessage;
};

/**
 * {@link WorkerOptionInputParam - 説明}
 */
 export type MessageOptionInputParam = ReplaceAnd<
    GetOptionObjectType<MessageOptionParam>,
    {
        //
    }
>;

/**
 * {@link WorkerOptionParam - 説明}
 */
export type MessageOptionParam = {
    room?: RoomType;
    worker?: WorkerType;
    reads?: ReadListType;
    reactions?: ReactionListType;
    threadHead?: ThreadHeadType;
    reply?: MessageType;
};

export type MessageType = MessageModel & MessageOptionParam;
export type GetMessageOptionParam = GetOptionParam<MessageType, MessageOptionParam, MessageOptionInputParam>;




export type MessageCLType = ReplaceAnd<
  MessageType,
  {
    room?: RoomCLType;
    worker?: WorkerCLType;
    reads?: ReadListCLType;
    reactions?: ReactionListCLType;
    threadHead?: ThreadHeadCLType;
    reply?: MessageCLType;
  } & CommonCLType
>;

export const toMessageCLType = (data?: MessageType): MessageCLType => {
  return {
    ...data,
    ...toCommonCLType(data),
    room: data?.room ? toRoomCLType(data?.room) : undefined,
    worker: data?.worker ? toWorkerCLType(data?.worker) : undefined,
    reads: data?.reads ? toReadListCLType(data?.reads) : undefined,
    reactions: data?.reactions ? toReactionListCLType(data?.reactions) : undefined,
    threadHead: data?.threadHead ? toThreadHeadCLType(data?.threadHead) : undefined,
    reply: data?.reply ? toMessageCLType(data?.reply) : undefined,
    
  } as MessageCLType;
};