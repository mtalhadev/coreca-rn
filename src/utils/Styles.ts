import { StyleSheet } from 'react-native'
import { THEME_COLORS } from './Constants'

export const FontStyle = {
    thin: 'NotoSansJP_100Thin',
    light: 'NotoSansJP_300Light',
    regular: 'NotoSansJP_400Regular',
    medium: 'NotoSansJP_500Medium',
    bold: 'NotoSansJP_700Bold',
    black: 'NotoSansJP_900Black',
}

export const GlobalStyles = StyleSheet.create({
    normalText: {
        fontSize: 14,
        fontFamily: FontStyle.regular,
        lineHeight: 16,
    },
    headerText: {
        fontSize: 20,
        fontFamily: FontStyle.bold,
        lineHeight: 22,
    },
    mediumText: {
        fontSize: 13,
        fontFamily: FontStyle.medium,
        lineHeight: 15,
    },
    boldText: {
        fontSize: 14,
        fontFamily: FontStyle.bold,
        lineHeight: 16,
    },
    smallText: {
        fontSize: 12,
        fontFamily: FontStyle.regular,
        lineHeight: 14,
    },
    smallGrayText: {
        fontSize: 12,
        fontFamily: FontStyle.regular,
        lineHeight: 14,
        color: THEME_COLORS.OTHERS.GRAY,
    },
})

export type ColorStyle = {
    mainColor: string
    subColor: string
    deepColor: string
    lightColor: string
    textColor: string
    deepTextColor: string
    highLightColor: string
    focusColor: string
    highLightTextColor: string
    menuHighlightColor: string
    anotherColorStyle: ColorStyle | undefined
}

export const BlueColor: ColorStyle = {
    mainColor: THEME_COLORS.BLUE.MIDDLE,
    subColor: THEME_COLORS.BLUE.MIDDLE_DEEP,
    deepColor: THEME_COLORS.BLUE.SUPER_DEEP,
    lightColor: THEME_COLORS.BLUE.SUPER_LIGHT,
    textColor: '#fff',
    deepTextColor: THEME_COLORS.BLUE.MIDDLE_DEEP,
    highLightColor: THEME_COLORS.BLUE.HIGH_LIGHT,
    focusColor: THEME_COLORS.BLUE.SUPER_DEEP,
    highLightTextColor: THEME_COLORS.BLUE.LIGHT,
    menuHighlightColor: THEME_COLORS.GREEN.LIGHT,
    anotherColorStyle: undefined,
}

export const GreenColor: ColorStyle = {
    mainColor: THEME_COLORS.GREEN.LIGHT,
    subColor: THEME_COLORS.GREEN.MIDDLE,
    deepColor: THEME_COLORS.GREEN.SUPER_DEEP,
    lightColor: THEME_COLORS.GREEN.SUPER_LIGHT,
    textColor: '#000',
    deepTextColor: THEME_COLORS.GREEN.DEEP,
    highLightColor: '#fff',
    focusColor: '#fff',
    highLightTextColor: THEME_COLORS.GREEN.MIDDLE,
    menuHighlightColor: THEME_COLORS.BLUE.LIGHT,
    anotherColorStyle: BlueColor,
}
BlueColor.anotherColorStyle = GreenColor
