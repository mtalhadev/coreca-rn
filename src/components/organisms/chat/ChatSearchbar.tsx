import React, { useState } from 'react';
import { View,StyleSheet, TextInput } from 'react-native';
import { SCREEN_WIDTH, THEME_COLORS } from '../../../utils/Constants';
import { BlueColor, FontStyle } from '../../../utils/Styles';
import { Icon } from '../../atoms/Icon';

export type Props = {
  onChangeText: (text: string) => void
}

export const ChatSearchbar = React.memo((props: Props) => {
  const { onChangeText } = props;
  const [text, setText] = useState<string>('');

  const updateText = (text: string) => {
    setText(text);
    onChangeText && onChangeText(text);
  };

  return (
      <View style={styles.searchBar}>
        <Icon name='filter' width={50} height={15} fill={BlueColor.mainColor}/>
        <TextInput
          style={{
              fontFamily: FontStyle.regular,
              fontSize: 14,
              lineHeight: 20,
              textAlignVertical: 'center',
              color: '#000',
              backgroundColor: 'transparent',
            }}
          placeholder={'検索'}
          placeholderTextColor={THEME_COLORS.OTHERS.LIGHT_GRAY}
          value={text}
          onChangeText={updateText}
      />
      </View>
  );
});

const styles = StyleSheet.create({
  searchBar: {
    height: 30,
    width: SCREEN_WIDTH * 0.9,
    justifyContent: 'flex-start',
    alignItems:'center',
    flexDirection: 'row',
		backgroundColor: 'transparent',
    borderBottomWidth: 1,
    borderBottomColor: '#555',
    marginTop: 10,
    marginBottom: 12,
    marginLeft: SCREEN_WIDTH * 0.1 * 0.5,
  },
});