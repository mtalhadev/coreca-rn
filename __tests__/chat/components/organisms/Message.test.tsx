import { render, screen } from '@testing-library/react-native';
import React from 'react';
import { Message } from '../../../../src/components/organisms/chat/Message';
import { CustomDate } from '../../../../src/models/_others/CustomDate';
import { MenuProvider } from 'react-native-popup-menu';
import { GreenColor } from '../../../../src/utils/Styles';

describe('Message test', () => {

    beforeEach(() => {
      jest.resetAllMocks()
    })
  
    it('renders Message type - text', () => {
      const date: CustomDate = new Date(2000,0,1,11,30,0).toCustomDate()

        render(
            <Message 
              message={{
                messageId: 'message-id',
                roomId: 'room-id',
                threadId: 'thread-id',
                message: "@TestUser It's a test",
                messageType: 'text',
                createdAt: date,
                worker: {
                  name: 'ABC',
                  xsImageUrl: 'https://profile-image-xs',
                  sImageUrl: 'https://profile-image-s',
                  imageUrl: 'https://profile-image',
                  imageColorHue: 40
                }
              }}
              myWorkerId={'my-worker-id'}
            />,
            {
              wrapper: MenuProvider
            }
        );

        expect(screen.getByText('ABC')).toBeVisible();
        expect(screen.getByText("@TestUser It's a test")).toBeVisible();
        expect(screen.getByText("11:30")).toBeVisible(); 
        const image = screen.getByTestId("ImageIcon-Image")
        expect(image.props.source.uri).toBe("https://profile-image-xs");
        expect(image.props.style.width).toBe(25);
        expect(image.props.style.height).toBe(25);
        const imageBorder = screen.getByTestId("ImageIcon-Border")
        expect(imageBorder.props.style.borderColor).toBe(GreenColor.subColor);       
        expect(screen).toMatchSnapshot();    
    }) 

    it('renders Message type - picture', () => {
      const date: CustomDate = new Date(2000,0,1,11,30,0).toCustomDate()

        render(
            <Message 
              message={{
                messageId: 'message-id',
                roomId: 'room-id',
                threadId: 'thread-id',
                message: "@TestUser It's a test",
                messageType: 'picture',
                attachmentUrl: 'https://attachment-picture',
                createdAt: date,
                worker: {
                  name: 'ABC',
                  xsImageUrl: 'https://profile-image-xs',
                  sImageUrl: 'https://profile-image-s',
                  imageUrl: 'https://profile-image',
                  imageColorHue: 40
                }
              }}
              myWorkerId={'my-worker-id'}
            />,
            {
              wrapper: MenuProvider
            }
        );

        expect(screen.getByText('ABC')).toBeVisible();
        expect(screen.getByText("11:30")).toBeVisible(); 
        const image = screen.getByTestId("ImageIcon-Image")
        expect(image.props.source.uri).toBe("https://profile-image-xs");
        expect(image.props.style.width).toBe(25);
        expect(image.props.style.height).toBe(25);
        const imagePicture = screen.getByTestId("message-picture")
        expect(imagePicture.props.source.uri).toBe("https://attachment-picture");
        const imageBorder = screen.getByTestId("ImageIcon-Border")
        expect(imageBorder.props.style.borderColor).toBe(GreenColor.subColor);       
        expect(screen).toMatchSnapshot();    
    }) 
         
    it('renders Message type - note', () => {
      const date: CustomDate = new Date(2000,0,1,11,30,0).toCustomDate()

        render(
            <Message 
              message={{
                messageId: 'message-id',
                roomId: 'room-id',
                threadId: 'thread-id',
                message: "@TestUser It's a test",
                messageType: 'note',
                createdAt: date,
                worker: {
                  name: 'ABC',
                  xsImageUrl: 'https://profile-image-xs',
                  sImageUrl: 'https://profile-image-s',
                  imageUrl: 'https://profile-image',
                  imageColorHue: 40
                }
              }}
              myWorkerId={'my-worker-id'}
            />,
            {
              wrapper: MenuProvider
            }
        );

        expect(screen.getByText('ABC')).toBeVisible();
        expect(screen.getByText("@TestUser It's a test")).toBeVisible();
        expect(screen.getByText("Notes")).toBeVisible();
        const image = screen.getByTestId("ImageIcon-Image")
        expect(image.props.source.uri).toBe("https://profile-image-xs");
        expect(image.props.style.width).toBe(25);
        expect(image.props.style.height).toBe(25);
        const imageBorder = screen.getByTestId("ImageIcon-Border")
        expect(imageBorder.props.style.borderColor).toBe(GreenColor.subColor);       
        expect(screen).toMatchSnapshot();    
    }) 
         
})  