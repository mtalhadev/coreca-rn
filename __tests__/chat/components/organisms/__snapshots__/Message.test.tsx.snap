// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message test renders Message type - note 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#aaa",
            "borderBottomWidth": 0.3,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#888",
                "borderBottomLeftRadius": 25,
                "borderBottomRightRadius": 25,
                "borderRadius": 25,
                "borderTopLeftRadius": 25,
                "borderTopRightRadius": 25,
                "height": 25,
                "justifyContent": "center",
                "marginTop": -4,
                "width": 25,
              }
            }
          >
            <RNCMaskedView>
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#000",
                      "borderRadius": 25,
                      "height": 25,
                      "width": 25,
                    }
                  }
                />
              </View>
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://profile-image-xs",
                  }
                }
                style={
                  Object {
                    "height": 25,
                    "width": 25,
                  }
                }
                testID="ImageIcon-Image"
              />
            </RNCMaskedView>
            <View
              style={
                Object {
                  "borderColor": "#C7D601",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "height": 25,
                  "position": "absolute",
                  "width": 25,
                }
              }
              testID="ImageIcon-Border"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
              "marginLeft": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "column",
                "paddingRight": 6,
              }
            }
          >
            <Text
              ellipsizeMode="middle"
              numberOfLines={1}
              style={
                Object {
                  "fontFamily": "NotoSansJP_700Bold",
                  "fontSize": 12,
                  "lineHeight": 14,
                  "width": 640,
                }
              }
            >
              ABC
            </Text>
            <Text
              style={
                Object {
                  "fontFamily": "NotoSansJP_400Regular",
                  "fontSize": 12,
                  "lineHeight": 16,
                  "marginTop": 6,
                  "width": 640,
                }
              }
            >
              <Text
                ellipsizeMode="middle"
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "fontFamily": "NotoSansJP_400Regular",
                      "fontSize": 12,
                      "lineHeight": 16,
                      "marginTop": 6,
                      "width": 640,
                    },
                    Object {
                      "color": "#0D6CB2",
                      "fontFamily": "NotoSansJP_700Bold",
                    },
                  ]
                }
              >
                @TestUser
              </Text>
              <Text
                ellipsizeMode="middle"
                style={
                  Array [
                    Object {
                      "fontFamily": "NotoSansJP_400Regular",
                      "fontSize": 12,
                      "lineHeight": 16,
                      "marginTop": 6,
                      "width": 640,
                    },
                    undefined,
                  ]
                }
              >
                 It's a test
              </Text>
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <View
            collapsable={false}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#606060",
                "borderRadius": 30,
                "borderWidth": 1,
                "elevation": 0,
                "height": 30,
                "justifyContent": "center",
                "marginLeft": -30,
                "marginTop": 2,
                "opacity": 1,
                "paddingHorizontal": 15,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 8,
              }
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": undefined,
                    "paddingHorizontal": undefined,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignContent": undefined,
                      "flexDirection": undefined,
                      "justifyContent": undefined,
                      "paddingTop": 0,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#343538",
                        "fontFamily": "NotoSansJP_500Medium",
                        "fontSize": 14,
                        "lineHeight": 30,
                        "paddingTop": 0,
                        "textAlign": undefined,
                      }
                    }
                  >
                    Notes
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;

exports[`Message test renders Message type - picture 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#aaa",
            "borderBottomWidth": 0.3,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#888",
                "borderBottomLeftRadius": 25,
                "borderBottomRightRadius": 25,
                "borderRadius": 25,
                "borderTopLeftRadius": 25,
                "borderTopRightRadius": 25,
                "height": 25,
                "justifyContent": "center",
                "marginTop": -4,
                "width": 25,
              }
            }
          >
            <RNCMaskedView>
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#000",
                      "borderRadius": 25,
                      "height": 25,
                      "width": 25,
                    }
                  }
                />
              </View>
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://profile-image-xs",
                  }
                }
                style={
                  Object {
                    "height": 25,
                    "width": 25,
                  }
                }
                testID="ImageIcon-Image"
              />
            </RNCMaskedView>
            <View
              style={
                Object {
                  "borderColor": "#C7D601",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "height": 25,
                  "position": "absolute",
                  "width": 25,
                }
              }
              testID="ImageIcon-Border"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
              "marginLeft": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "column",
                "paddingRight": 6,
              }
            }
          >
            <Text
              ellipsizeMode="middle"
              numberOfLines={1}
              style={
                Object {
                  "fontFamily": "NotoSansJP_700Bold",
                  "fontSize": 12,
                  "lineHeight": 14,
                  "width": 640,
                }
              }
            >
              ABC
            </Text>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <Image
                source={
                  Object {
                    "uri": "https://attachment-picture",
                  }
                }
                style={
                  Object {
                    "borderRadius": 5,
                    "height": 450,
                    "marginTop": 6,
                    "resizeMode": "contain",
                    "width": 450,
                  }
                }
                testID="message-picture"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginTop": 8,
                "width": 640,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#737680",
                  "fontSize": 10,
                  "marginTop": 20,
                }
              }
            >
              11:30
            </Text>
            <View
              style={
                Object {
                  "marginTop": 10,
                }
              }
            >
              <View
                collapsable={false}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    ctx={
                      Object {
                        "menuActions": Object {
                          "_getOpenedMenu": [Function],
                          "_notify": [Function],
                          "closeMenu": [Function],
                          "isMenuOpen": [Function],
                          "openMenu": [Function],
                          "toggleMenu": [Function],
                        },
                        "menuRegistry": Object {
                          "getAll": [Function],
                          "getMenu": [Function],
                          "setOptionsCustomStyles": [Function],
                          "subscribe": [Function],
                          "unsubscribe": [Function],
                          "updateLayoutInfo": [Function],
                        },
                      }
                    }
                    style={
                      Array [
                        Object {
                          "top": -42,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;

exports[`Message test renders Message type - text 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#aaa",
            "borderBottomWidth": 0.3,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#888",
                "borderBottomLeftRadius": 25,
                "borderBottomRightRadius": 25,
                "borderRadius": 25,
                "borderTopLeftRadius": 25,
                "borderTopRightRadius": 25,
                "height": 25,
                "justifyContent": "center",
                "marginTop": -4,
                "width": 25,
              }
            }
          >
            <RNCMaskedView>
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#000",
                      "borderRadius": 25,
                      "height": 25,
                      "width": 25,
                    }
                  }
                />
              </View>
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://profile-image-xs",
                  }
                }
                style={
                  Object {
                    "height": 25,
                    "width": 25,
                  }
                }
                testID="ImageIcon-Image"
              />
            </RNCMaskedView>
            <View
              style={
                Object {
                  "borderColor": "#C7D601",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "height": 25,
                  "position": "absolute",
                  "width": 25,
                }
              }
              testID="ImageIcon-Border"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
              "marginLeft": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "column",
                "paddingRight": 6,
              }
            }
          >
            <Text
              ellipsizeMode="middle"
              numberOfLines={1}
              style={
                Object {
                  "fontFamily": "NotoSansJP_700Bold",
                  "fontSize": 12,
                  "lineHeight": 14,
                  "width": 640,
                }
              }
            >
              ABC
            </Text>
            <Text
              style={
                Object {
                  "fontFamily": "NotoSansJP_400Regular",
                  "fontSize": 12,
                  "lineHeight": 16,
                  "marginTop": 6,
                  "width": 640,
                }
              }
            >
              <Text
                ellipsizeMode="middle"
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "fontFamily": "NotoSansJP_400Regular",
                      "fontSize": 12,
                      "lineHeight": 16,
                      "marginTop": 6,
                      "width": 640,
                    },
                    Object {
                      "color": "#0D6CB2",
                      "fontFamily": "NotoSansJP_700Bold",
                    },
                  ]
                }
              >
                @TestUser
              </Text>
              <Text
                ellipsizeMode="middle"
                style={
                  Array [
                    Object {
                      "fontFamily": "NotoSansJP_400Regular",
                      "fontSize": 12,
                      "lineHeight": 16,
                      "marginTop": 6,
                      "width": 640,
                    },
                    undefined,
                  ]
                }
              >
                 It's a test
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginTop": 8,
                "width": 640,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#737680",
                  "fontSize": 10,
                  "marginTop": 20,
                }
              }
            >
              11:30
            </Text>
            <View
              style={
                Object {
                  "marginTop": 10,
                }
              }
            >
              <View
                collapsable={false}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    ctx={
                      Object {
                        "menuActions": Object {
                          "_getOpenedMenu": [Function],
                          "_notify": [Function],
                          "closeMenu": [Function],
                          "isMenuOpen": [Function],
                          "openMenu": [Function],
                          "toggleMenu": [Function],
                        },
                        "menuRegistry": Object {
                          "getAll": [Function],
                          "getMenu": [Function],
                          "setOptionsCustomStyles": [Function],
                          "subscribe": [Function],
                          "unsubscribe": [Function],
                          "updateLayoutInfo": [Function],
                        },
                      }
                    }
                    style={
                      Array [
                        Object {
                          "top": -42,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
